cmd/collector/app/zipkin/json.go

```diff
@@ -0,0 +1,236 @@
+	var tSpans []*zipkincore.Span
+	for _, span := range spans {
+		tSpan, err := spanToThrift(span)
+		if err != nil {
+			return nil, err
+		}
+		tSpans = append(tSpans, tSpan)
+	}
+	return tSpans, nil
+}
+
+func spanToThrift(s zipkinSpan) (*zipkincore.Span, error) {
+	id, err := hexToUnsignedLong(s.ID)
+	if err != nil {
+		return nil, err
+	}
+	traceID, err := hexToUnsignedLong(s.TraceID)
+	if err != nil {
+		return nil, err
+	}
+
+	tSpan := &zipkincore.Span{
+		ID:        int64(id),
+		TraceID:   int64(traceID),
+		Name:      s.Name,
+		Debug:     s.Debug,
+		Timestamp: s.Timestamp,
+		Duration:  s.Duration,
+	}
+
+	if len(s.TraceID) == 32 {
+		// take 0-16
+		traceIDHigh, err := hexToUnsignedLong(s.TraceID[:16])
+		if err != nil {
+			return nil, err
+		}
+		help := int64(traceIDHigh)
+		tSpan.TraceIDHigh = &help
+	}
+
+	if len(s.ParentID) > 0 {
+		parentID, err := hexToUnsignedLong(s.ParentID)
+		if err != nil {
+			return nil, err
+		}
+		signed := int64(parentID)
+		tSpan.ParentID = &signed
+	}
+
+	for _, a := range s.Annotations {
+		tA, err := annoToThrift(a)
+		if err != nil {
+			return nil, err
+		}
+		tSpan.Annotations = append(tSpan.Annotations, tA)
+	}
+	for _, ba := range s.BinaryAnnotations {
+		tBa, err := binAnnoToThrift(ba)
+		if err != nil {
+			return nil, err
+		}
+		tSpan.BinaryAnnotations = append(tSpan.BinaryAnnotations, tBa)
+	}
+
+	return tSpan, nil
+}
+
+func endpointToThrift(e endpoint) (*zipkincore.Endpoint, error) {
+	ipv4, err := parseIpv4(e.IPv4)
+	if err != nil {
+		return nil, err
+	}
+
+	return &zipkincore.Endpoint{
+		ServiceName: e.ServiceName,
+		Port:        e.Port,
+		Ipv4:        ipv4,
+		Ipv6:        []byte(e.IPv6),
+	}, nil
+}
+
+func annoToThrift(a annotation) (*zipkincore.Annotation, error) {
+	endpoint, err := endpointToThrift(a.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return &zipkincore.Annotation{
+		Timestamp: a.Timestamp,
+		Value:     a.Value,
+		Host:      endpoint,
+	}, nil
+}
+
+func binAnnoToThrift(ba binaryAnnotation) (*zipkincore.BinaryAnnotation, error) {
+	endpoint, err := endpointToThrift(ba.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return &zipkincore.BinaryAnnotation{
+		Key:            ba.Key,
+		Value:          []byte(ba.Value),
+		Host:           endpoint,
+		AnnotationType: zipkincore.AnnotationType_STRING,
+	}, nil
+}
+
+func parseIpv4(str string) (int32, error) {
+	segments := strings.Split(str, ".")
+	if len(segments) == 1 {
+		return 0, nil
+	}
+
+	var ipv4 int32
+	for _, segment := range segments {
+		parsed, err := strconv.ParseInt(segment, 10, 32)
+		if err != nil {
+			return 0, errWrongIpv4
+		}
+		parsed32 := int32(parsed)
+		ipv4 = ipv4<<8 | (parsed32 & 0xff)
+	}
+
+	return ipv4, nil
+}
+
+func hexToUnsignedLong(hex string) (uint64, error) {
+	len := len(hex)
+	if len < 1 || len > 32 {
+		return 0, errIsNotUnsignedLog
+	}
+
+	start := 0
+	if len > 16 {
+		start = len - 16
+	}
+
+	var result uint64
+	for i := start; i < len; i++ {
```

Comment:
why not use model functions like `model/TraceIDFromString`? 

https://github.com/uber/jaeger/blob/master/model/span.go#L146

It automatically takes care of high/low.